lib/macho/macho_file.rb

```diff
@@ -363,13 +363,16 @@ def rpaths
     # @param new_path [String] the new runtime path
     # @return [void]
     # @raise [MachO::RpathUnknownError] if no such old runtime path exists
+    # @raise [MachO::RpathExistsError] if the new runtime path already exists
     def change_rpath(old_path, new_path)
       old_lc = command(:LC_RPATH).find { |r| r.path.to_s == old_path }
       raise RpathUnknownError.new(old_path) if old_lc.nil?
+      raise RpathExistsError.new(new_path) if rpaths.include?(new_path)
 
       new_lc = LoadCommand.create(:LC_RPATH, new_path)
 
-      replace_command(old_lc, new_lc)
+      delete_rpath(old_path)
+      insert_command(old_lc.view.offset, new_lc)
```

Comment:
Clever! I was looking forward to some fiddling with `replace_command` and `delete_command`. :wink:

The only concern that I have, because we also haven't made this assumption in `delete_rpath`, is that the first match will be automatically the first one inside the Mach-O file (in terms of file offset).